internal/integration_test/engine/adhoc_test.go

```diff
@@ -62,10 +63,15 @@ var tests = map[string]testCase{
 	"before listener stack iterator":                                   {f: testBeforeListenerStackIterator},
 	"before listener stack iterator offsets":                           {f: testListenerStackIteratorOffset, wazevoSkip: true},
 	"many params many results / doubler":                               {f: testManyParamsResultsDoubler},
+	"many params many results / doubler / listener":                    {f: testManyParamsResultsDoubler_listener},
 	"many params many results / call_many_consts":                      {f: testManyParamsResultsCallManyConsts},
+	"many params many results / call_many_consts / listener":           {f: testManyParamsResultsCallManyConsts_listener, wazevoSkip: true},
 	"many params many results / swapper":                               {f: testManyParamsResultsSwapper},
+	"many params many results / swapper / listener":                    {f: testManyParamsResultsSwapper_listener},
 	"many params many results / main":                                  {f: testManyParamsResultsMain},
+	"many params many results / main / listener":                       {f: testManyParamsResultsMain_listener},
 	"many params many results / call_many_consts_and_pick_last_vector": {f: testManyParamsResultsCallManyConstsAndPickLastVector},
+	"many params many results / call_many_consts_and_pick_last_vector / listener": {f: testManyParamsResultsCallManyConstsAndPickLastVector_listener, wazevoSkip: true},
```

Comment:
finally passed all the many params/results tests with or without function listeners!!!!!!1